### ISSUE TITLE
CI: system tests: start_registry misbehaving

### ISSUE THREAD

**Issue by edsantiago** [containers/podman#23725]
Weird one:
```
[+0554s] not ok 240 [260] podman pull - EXTEND_TIMEOUT_USEC in 2446ms
...
<+     > # # podman [registry opts] network reload --all
<+824ms> # 4aebed84cef198f49bcaae5696ffb7fc812238ac1beda008068deddffc854ae0
         #
<+018ms> # # podman push --tls-verify=false --creds user7V6g:pwZX1laz7cDjf1Z4M quay.io/libpod/testimage:20240123 localhost:42564/name:tag
<+121ms> # Getting image source signatures
         # Copying blob sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef
         # Copying blob sha256:25a964f60e84b2d24f8098376c985a5c057bea74d699ee50c93e417c26516cc3
         # Copying config sha256:1f6acd4c4a1d4d39395870e1c12dbb68da27dd432afecf86e7cb6e23cf4b75d0
         # Writing manifest to image destination
         #
<+011ms> # # podman pull --tls-verify=false --creds user7V6g:pwZX1laz7cDjf1Z4M localhost:42564/name:tag
<+049ms> # Trying to pull localhost:42564/name:tag...
         # Error: dial unixgram /tmp/CI_a4XK/podman_bats.XbmZm2/notify.sock: connect: no such file or directory
<+005ms> # [ rc=125 (** EXPECTED 0 **) ]
```
Seeing something similar in a quadlet test too:
```
<+     > # # podman [...] network reload --all
         #
<+133ms> # # podman login --authfile=/tmp/CI_Ia6y/podman_bats.cgIxJ5/authfile.json --tls-verify=false --username userfzbg --password pwnHxU9N3L2sFAVj3 localhost:42412
<+076ms> # Error: authenticating creds for "localhost:42412": pinging container registry localhost:42412: Get "http://localhost:42412/v2/": dial tcp [::1]:42412: connect: connection refused
```
This is NOT IN PARALLEL. It is in serial system tests, run prior to parallel, but it is only happening in my parallel-testing PR.

It is possible that these are two separate issues, and I'm being silly by treating them as one because of the `podman network reload`. For now, this is my bug to deal with, I'm collecting in one place for convenience, and I'll fork into two later if necessary. 

* debian-13 : sys podman debian-13 root host sqlite
  * PR #23275
    * [08-22 08:08](https://api.cirrus-ci.com/v1/artifact/task/5271340641419264/html/sys-podman-debian-13-root-host-sqlite.log.html#t--00240) in [sys] [260] podman pull - EXTEND_TIMEOUT_USEC
    * [08-20 21:58](https://api.cirrus-ci.com/v1/artifact/task/5509423630647296/html/sys-podman-debian-13-root-host-sqlite.log.html#t--00240) in [sys] [260] podman pull - EXTEND_TIMEOUT_USEC
    * [08-14 15:45](https://api.cirrus-ci.com/v1/artifact/task/4678865774182400/html/sys-podman-debian-13-root-host-sqlite.log.html#t--00281) in [sys] [260] podman pull - EXTEND_TIMEOUT_USEC
* fedora-40 : sys podman fedora-40 rootless host sqlite
  * PR #23275
    * [08-15 17:22](https://api.cirrus-ci.com/v1/artifact/task/6670988945915904/html/sys-podman-fedora-40-rootless-host-sqlite.log.html#t--00154p) in [sys] |252| quadlet - image files
* fedora-40-aarch64 : sys podman fedora-40-aarch64 root host sqlite
  * PR #23275
    * [08-22 08:08](https://api.cirrus-ci.com/v1/artifact/task/6397240548261888/html/sys-podman-fedora-40-aarch64-root-host-sqlite.log.html#t--00195p) in [sys] |252| quadlet - image files

| x | x | x | x | x | x |
| ---: | ---: | ---: | ---: | ---: | ---: |
| sys(5) | podman(5) | debian-13(3) | root(4) | host(5) | sqlite(5) |
|  |  | fedora-40(1) | rootless(1) |  |  |
|  |  | fedora-40-aarch64(1) |  |  |  |

**Comment by Luap99** [comment]
Weird, podman network reload --all is not parallel safe as it removes all interfaces/firewall rules for the container so all current connections are interrupted leading to weird errors. Although this should not matter for the rootless case here as it uses pasta which doesn't do network reload at all.
But if this happens in the not parallel part that is definitely odd.

**Comment by edsantiago** [comment]
I'm going to assume this was fixed by #24002
